spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+describe SessionsController do
+
+  describe 'Get new' do
+    context 'with authenticated user' do 
+      it 'renders the new template' do 
+        session[:user_id] = nil
+        get :new
```

Comment:
I wouldn't test this, it is fairly standard behavior if you `get :new` to `render :new`. I know you have written a control flow  but I would only test positive happy path for the `if` or `unless`. After all - do not test code you have not written.